netket/dynamics.py

```diff
@@ -1 +1,362 @@
-from . import _core
+# Copyright 2020 The Simons Foundation, Inc. - All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as _np
+import scipy
+
+from netket.vmc_common import tree_map, trees2_map
+
+from netket.abstract_variational_driver import AbstractVariationalDriver
+from netket._vmc import Vmc
+from netket._steadystate import SteadyState
+
+from netket.machine.density_matrix import AbstractDensityMatrix
+
+from .operator import (
+    local_values as _local_values,
+    der_local_values as _der_local_values,
+)
+from netket.stats import (
+    statistics as _statistics,
+    mean as _mean,
+    sum_inplace as _sum_inplace,
+)
+
+from functools import singledispatch
+
+from ._scipy_integrators import EulerSolver
+
+
+@singledispatch
+def _time_evo(self):
+    raise ErrorException("unknown driver type {}".format(type(self)))
```

Comment:
This should probably another error type, probably `ValueError`?
```suggestion
    raise TypeError("unknown driver type {}".format(type(self)))
```